controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -126,24 +117,75 @@ protected void startUp() throws InterruptedException, Exception {
+                if (controllerEventProcessors.isRunning()) {
+                    log.info("handling host removed and reporting readers offline for host {}", host.getHostId());
+                    return controllerEventProcessors.notifyProcessFailure(host.getHostId());
+                } else {
+                    return CompletableFuture.completedFuture(null);
+                }
+            }, e -> log.warn(e.getMessage()), executor);
+        });
+
+        txnSweeperOpt.ifPresent(txnSweeper -> {
+            RetryHelper.withIndefiniteRetriesAsync(() -> {
+                if(txnSweeper.isRunning()) {
+                    log.info("Sweeping orphaned transactions for host {}", host.getHostId());
+                    return txnSweeper.sweepOrphanedTxns(host.getHostId());
+                } else {
+                    return CompletableFuture.completedFuture(null);
+                }
+            }, e -> log.warn(e.getMessage()), executor);
+        });
+    }
 
-            if (eventProcessorsOpt.isPresent()) {
+    private void sweepEventProcessorReaders(Supplier<Set<String>> processes) {
+        eventProcessorsOpt.ifPresent(eventProcessors -> {
+            // Await initialization of eventProcesorsOpt
+            RetryHelper.withIndefiniteRetriesAsync(() -> {
+                log.info("Awaiting controller event processors' start");
+                eventProcessors.awaitRunning();
                 // Sweep orphaned tasks or readers at startup.
                 log.info("Sweeping orphaned readers at startup");
-                eventProcessorsOpt.get().handleOrphanedReaders(processes);
-            }
+                return eventProcessors.handleOrphanedReaders(processes);
+            }, e -> log.warn(e.getMessage()), executor);
+        });
+    }
 
-            if (txnSweeperOpt.isPresent()) {
+    private void sweepTransactions(Supplier<Set<String>> processes) {
+        txnSweeperOpt.ifPresent(txnSweeper -> {
+            // Await initialization of transactionTasksOpt.
+            RetryHelper.withIndefiniteRetriesAsync(() -> {
+                log.info("Awaiting StreamTransactionMetadataTasks to get ready");
+                Exceptions.handleInterrupted(txnSweeper::awaitInitialization);
                 // Sweep orphaned transactions as startup.
                 log.info("Sweeping orphaned transactions");
-                txnSweeperOpt.get().sweepFailedHosts(processes);
-            }
+                return txnSweeper.sweepFailedHosts(processes);
+            }, e -> log.warn(e.getMessage()), executor);
+        });
+    }
 
-            log.info("Controller cluster listener startUp complete");
-        } finally {
-            LoggerHelpers.traceLeave(log, objectId, "startUp", traceId);
-        }
+    private void sweepTasks(Supplier<Set<String>> processes) {
+        log.info("Sweeping orphaned tasks at startup");
+        RetryHelper.withIndefiniteRetriesAsync(() -> taskSweeper.sweepOrphanedTasks(processes),
```

Comment:
not all code paths inside sweepers are covered with retries and exceptions can still be thrown. 
We dont want to be in a situation that because of any error we are not sweeping failed hosts. 
Will remove redundant retry blocks inside sweepers in a separate issue. created issue 1569